src/colmap/util/endian.h

```diff
@@ -131,6 +131,26 @@ T NativeToBigEndian(const T x) {
   }
 }
 
+template <typename T1, typename T2>
+T2 LittleEndianToNativeCast(const T1 x) {
```

Comment:
I suggest to not introduce these new helpers but instead apply the static_cast explicitly on the call site.